### PR TITLE
[docs] Fix dead links

### PR DESCRIPTION
**Pull request by Harshg999**
## What changes were proposed in this pull request?

- Change few dead links in documentation flagged by CI checks

## How was this patch tested?

- Manually and existing documentation CI check

### CODE REVIEW

File: docs/docs-site/content/administrator/configuration/connectors/_index.md [hunk 1]
```diff
@@ -758,7 +758,7 @@ Alternative:
 
 ### Dask SQL
 
-[dask-sql](https://dask-sql.readthedocs.io) adds a SQL layer on top of Dask.
+dask-sql adds a SQL layer on top of Dask.
```
Comment (agl29) [inline docs/docs-site/content/administrator/configuration/connectors/_index.md:761]: can we use this link ->`https://dask-sql.readthedocs.io/en/latest/` ?

File: docs/docs-site/content/developer/development/_index.md [hunk 2]
```diff
@@ -508,7 +508,7 @@ Run the user interface tests:
 
 In the pyproject.toml file, we define the configuration for pytest, including settings to enable pytest to discover and execute the unit tests.
 
-For pytest to recognize test files, you should follow the naming convention `*_test.py` or `test_*.py`. Following the pytest framework [convention](https://docs.pytest.org/en/7.1.x/explanation/goodpractices.html#conventions-for-python-test-discovery) for test discovery is recommended. Pytest will gather test items according to the following rules:
```
Comment (agl29) [inline docs/docs-site/content/developer/development/_index.md:511]: any specific reason to remove this link?

File: docs/docs-site/content/developer/development/_index.md [hunk 3]
```diff
@@ -820,8 +820,6 @@ A Pypi token will be needed. For doing a test release https://test.pypi.org/proj
     rm -rf dist && python3 -m build && python3 -m twine upload --repository testpypi dist/*
     python3 -m pip install --index-url https://test.pypi.org/simple/ gethue --upgrade
 
-Read more on [Python packaging](https://packaging.python.org/tutorials/packaging-projects/#generating-distribution-archives).
```
Comment (agl29) [inline docs/docs-site/content/developer/development/_index.md:823]: why we are removing this link? is this not working or not relevant?

File: docs/docs-site/content/administrator/configuration/connectors/_index.md (hunk 1 above)
Comment (Harshg999) [reply docs/docs-site/content/administrator/configuration/connectors/_index.md:761]: Its giving 403 error

File: docs/docs-site/content/developer/development/_index.md (hunk 2 above)
Comment (Harshg999) [reply docs/docs-site/content/developer/development/_index.md:511]: Its giving 403 error

File: docs/docs-site/content/developer/development/_index.md (hunk 3 above)
Comment (Harshg999) [reply docs/docs-site/content/developer/development/_index.md:823]: Its giving 403 error

**Review by bjornalm — APPROVED** [review]

**Review by agl29 — APPROVED** [review]

**Comment by Harshg999** [comment]
We need to find a long term solution for this dead link check - either remove it completely or find an alternate or fix the check to work for all links.

**Pull request merged**